src/scikit_build_core/resources/_editable_redirect.py

```diff
@@ -92,13 +115,17 @@ def find_spec(
                 fullname,
                 os.path.join(self.dir, redir),
                 submodule_search_locations=submodule_search_locations,
+                loader=ScikitBuildRedirectingLoader(
+                    fullname, os.path.join(self.dir, redir)
+                ),
             )
         if fullname in self.known_source_files:
             redir = self.known_source_files[fullname]
             return importlib.util.spec_from_file_location(
                 fullname,
                 redir,
                 submodule_search_locations=submodule_search_locations,
+                loader=ScikitBuildRedirectingLoader(fullname, redir),
```

Comment:
The fact that this is specifying a single directory (the redir) is perhaps part of where we could improve things. (Sub)packages are found in the known source files list by their `__init__.py`. As a result, that path is used as the root, which means that we cannot find files nested in the package but present in the build directory because the package does not exist in `known_wheel_files`.